### PR TITLE
[STK-188][STYLES] - Center dApp background image and update date pickers spacing

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Fixes: [STK-188](https://linear.app/swaprhq/issue/STK-188/cosmetic-fixes)

# Description
* Moved some code out from the Stackbox and into SelectNetwork (concern separation)
* Created custom code to handle the Chain switch as app state 
* Updated the way we consume the chain switch where it's needed

# Visual evidence
## Date picker Spacing, Figma File:
image
image

## Date picker Spacing, Old Version:
image

## Date picker Spacing, New Version:
image

## Background image horizontal centered, Old Version:
image

## Background image horizontal centered, New Version:
image

# How to test the changes

1) Pull this branch
2) Run the project locally
3) Go to Stackly dApp
4) Connect your wallet
5) 
    a. The matrix background image should be both horizontally and vertically centered
    b. The container for both date pickers should have the correct spacing for both mobile and desktop

### CODE REVIEW

**Review by Diogomartf — APPROVED** [review]

File: packages/app/components/stackbox/Stackbox.tsx [hunk 1]
```diff
@@ -475,7 +475,7 @@ export const Stackbox = () => {
             </div>
             <div className="space-y-1">
               <div className="flex flex-col border divide-y md:divide-y-0 md:flex-row rounded-2xl border-surface-50 md:divide-x divide-surface-50">
-                <div className="flex flex-col w-full p-3 space-y-2 hover:bg-surface-25">
+                <div className="flex flex-col w-full pl-4 pr-3 py-2 lg:py-3 space-y-2 hover:bg-surface-25">
```
Comment (berteotti) [inline packages/app/components/stackbox/Stackbox.tsx:478]: This changes make the `DatePicker` input strange on mobile and tablet. With different padding from the other input.

Comment (berteotti) [inline packages/app/components/stackbox/Stackbox.tsx:478]: This changes make the DatePicker input strange on mobile and tablet. With different padding from the other input.

**Review by berteotti — CHANGES_REQUESTED** [review]
This style improvement should be done before merging.

File: packages/app/components/stackbox/Stackbox.tsx (hunk 1 above)
Comment (Diogomartf) [reply packages/app/components/stackbox/Stackbox.tsx:478]: what's the reason to change `padding` here?

**Comment by Diogomartf** [comment]
I think the description is not of this PR. 

The screenshots are not that clear on what's the difference. Maybe it could be the difference between date pickers.

**Comment by ElRodrigote** [comment]
> I think the description is not of this PR.
> 
> The screenshots are not that clear on what's the difference. Maybe it could be the difference between date pickers.

I'll improve the screenshots and change into a more descriptive title. Thanks!

File: packages/app/components/stackbox/Stackbox.tsx (hunk 1 above)
Comment (ElRodrigote) [reply packages/app/components/stackbox/Stackbox.tsx:478]: It was requested in the ticket, it was one of the two updates to be done requested by Akash and prioritized by Venky
https://linear.app/swaprhq/issue/STK-188/cosmetic-fixes

**Review by berteotti — APPROVED** [review]
Looks good to me!

**Pull request merged**